lib/mongoid/traversable.rb

```diff
@@ -178,7 +178,7 @@ def collect_descendants
         to_expand = []
         expanding.each do |child|
           next if expanded[child]
-          expanded[child] = true
+          expanded[child] = true if child.id
```

Comment:
The idea here is that documents are compared by their `id`. If the `id` for multiple documents is `nil`, then they will all compare equally. In this case, if multiple documents had a `nil` `id`, they were being dropped by the line above this one.

Reply:
We should prefer `._id` to `.id`